### DISCUSSION TITLE
Focus the last window after closing a terminal split.

### DISCUSSION THREAD

**Discussion by Filip62** [folke/snacks.nvim#153] [Q&A]
Hello.

While working in some window, when I open a terminal split, and close it, the window that gets focused is not the one I toggled the terminal from. If there are at least two windows, toggling behaves like this in one of them.

This doesn't happen with floating terminals, I've switched to floating terms for now.

Is there a way to configure this behavior?

Thanks.

By toggling, I mean just calling `require("snacks").terminal()` twice in a row.

**Comment by dpetka2001** [comment]
This should do what you want
```lua
{
  "folke/snacks.nvim",
  opts = {
    styles = {
      terminal = {
        keys = {
            term_quit = {
              "<C-q>",
              function(self)
                local alt_win = vim.fn.win_getid(vim.fn.winnr("#"))

                self:hide()

                vim.schedule(function()
                  if alt_win and vim.api.nvim_win_is_valid(alt_win) then
                    vim.api.nvim_set_current_win(alt_win)
                  end
                end)
              end,
              mode = "t",
              desc = "Close terminal and focus last window",
            },
        },
      },
    },
  },
}
```
Test it out. I chose `<C-q>` because this will just hide the window and when you re-open it it will continue the session. I left `<C-d>` untouched because I personally still want to use `<C-d>` to completely exit the terminal and not save state and have a different keymap for just hiding the terminal without destroying the session, but you can choose whatever keymap works best for your needs.

**Reply by Filip62** [reply]
Works just fine, thank you.

I have a keymap that just calls `require("snacks").terminal(...)`, and that same keymap is overridden in the style configuration like you showed me here and it is exactly what I wanted, a toggle that switches to the last focused window on `:hide()` basically.

**Comment by folke** [comment]
fyi, snacks windows now automatically go back to the last window
